### PR TITLE
CCM-5100: Backport suggested changes to template repo

### PR DESCRIPTION
**Pull request by m-houston**
## Description

During development on the IAM AuthN PoC several issues were encountered running the scripts from the template repo, both pre-commit hooks and deployment related.

This PR contains the proposed changes which may be useful to include in the template.

* Update terraform version to 1.8.5
* Init terraform environments before validating
* Fix _terraform helper function in Makefile for non-bash shells
* Remove hidden directories from vscode config
* Change hooks to use branch checks rather than just staged changes
* Spelling and dictionary tweaks

## Context

* Running on M1 Mac
* Should be tested on Windows

## Type of changes

- [ ] Refactoring (non-breaking change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [x] I have followed the code style of the project
- [ ] I have added tests to cover my changes
- [ ] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [x] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

File: scripts/config/pre-commit.yaml
Comment (RossBugginsNHS) [inline scripts/config/pre-commit.yaml:1]: My change to /bin/env felt a bit off - assume on mac env is only in /usr/bin ? 

Can we think of any other options for doing this?

**Review by RossBugginsNHS — COMMENTED** [review]
Is there any better option than what we've done to set the envvar using /usr/bin/env? Not sure if there is a better option? are we going to flip flop on this if this doesn't work elsewhere?

File: scripts/config/pre-commit.yaml
Comment (timireland) [reply scripts/config/pre-commit.yaml:1]: Set and unset  ?  seems clunky

Comment (RossBugginsNHS) [reply scripts/config/pre-commit.yaml:1]: > Set and unset ? seems clunky

example code?

File: scripts/terraform/terraform.mk [hunk 1]
```diff
@@ -47,8 +47,8 @@ clean:: # Remove Terraform files (terraform) - optional: terraform_dir|dir=[path
 
 _terraform: # Terraform command wrapper - mandatory: cmd=[command to execute]; optional: dir=[path to a directory where the command will be executed, relative to the project's top-level directory, default is one of the module variables or the example directory, if not set], opts=[options to pass to the Terraform command, default is none/empty]
 	dir=$(or ${dir}, ${TERRAFORM_STACK})
-	source scripts/terraform/terraform.lib.sh
-	terraform-${cmd} # 'dir' and 'opts' are accessible by the function as environment variables, if set
+	. scripts/terraform/terraform.lib.sh && \
```
Comment (aidenvaines-bjss) [inline scripts/terraform/terraform.mk:50]: source isnt posix where as . is, and make is fussy. nice spot, this one took me a while to find on my -internal branch.

**Review by aidenvaines-bjss — APPROVED** [review]

File: scripts/config/pre-commit.yaml
Comment (m-houston) [reply scripts/config/pre-commit.yaml:1]: Will go with this for now until something better is available

**Review by RossBugginsNHS — APPROVED** [review]

**Pull request merged**